h/views/api/annotations.py

```diff
@@ -190,18 +190,31 @@ def _annotation_resource(request, annotation):
 
 def _record_search_api_usage_metrics(
     params,
-    record_metrics=newrelic.agent.record_custom_metrics,
+    record_param=newrelic.agent.add_custom_parameter,
 ):
-    metrics = [
-        # Record usage of deprecated offset.
-        ('Custom/SearchApi/offset', int("offset" in params)),
-
+    # Record usage of search params and associate them with a transaction.
+    keys = [
+        # Record usage of inefficient offset and it's alternative search_after.
+        "offset",
+        "search_after",
+        "sort",
         # Record usage of url/uri (url is an alias of uri).
-        ('Custom/SearchApi/url', int("url" in params)),
-        ('Custom/SearchApi/uri', int("uri" in params)),
-
+        "url",
+        "uri",
         # Record usage of tags/tag (tags is an alias of tag).
-        ('Custom/SearchApi/tags', int("tags" in params)),
-        ('Custom/SearchApi/tag', int("tag" in params)),
+        "tags",
+        "tag",
+        # Record usage of _separate_replies which will help distinguish client calls
+        # for loading the sidebar annotations from other api calls.
+        "_separate_replies",
+        # Record group and user-these help in identifying slow queries.
+        "group",
+        "user",
+        # Record usage of wildcard feature.
+        "wildcard_uri",
     ]
-    record_metrics(metrics)
+
+    for k in keys:
+        if k in params:
+            # The NewRelic query language does not permit _ at the begining.
```

Comment:
Is there not a way to quote fields that start with "_"?

Reply:
There is a quote option using back tick but the _ key wasn't even showing up as an attribute in the ui. I'm not sure if that's a bug in the ui/attribute collector or that _ are reserved/not permitted in attribute/parameter names.

Reply:
I don't see anything about _ in the [docs](https://docs.newrelic.com/docs/insights/insights-data-sources/custom-data/insights-custom-data-requirements-limits#reserved-words). I do see something about offset tho-perhaps I should change that not that we use offset in our queries or that I anticipate that we will but it's an easy enough fix.